pkg/pmm/container.go

```diff
@@ -28,21 +29,65 @@ func Container(
 		})
 	}
 
-	return corev1.Container{
+	container := corev1.Container{
 		Name:            "pmm-client",
 		Image:           pmmSpec.Image,
 		ImagePullPolicy: pmmSpec.ImagePullPolicy,
 		SecurityContext: pmmSpec.ContainerSecurityContext,
 		Ports:           ports,
 		Resources:       pmmSpec.Resources,
 		Env:             envs,
+		Lifecycle: &corev1.Lifecycle{
+			PreStop: &corev1.LifecycleHandler{
+				Exec: &corev1.ExecAction{
+					Command: []string{
+						"bash",
+						"-c",
+						"pmm-admin unregister --force",
+					},
+				},
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			InitialDelaySeconds: 60,
+			TimeoutSeconds:      5,
+			PeriodSeconds:       10,
+			ProbeHandler: corev1.ProbeHandler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port: intstr.FromInt32(7777),
+					Path: "/local/Status",
+				},
+			},
+		},
 		VolumeMounts: []corev1.VolumeMount{
 			{
 				Name:      apiv1alpha1.BinVolumeName,
 				MountPath: apiv1alpha1.BinVolumePath,
 			},
 		},
 	}
+
+	if pmmSpec.LivenessProbes != nil {
+		container.LivenessProbe = pmmSpec.LivenessProbes
+		if reflect.DeepEqual(container.LivenessProbe.ProbeHandler, corev1.ProbeHandler{}) {
+			container.LivenessProbe.HTTPGet = &corev1.HTTPGetAction{
+				Port: intstr.FromInt32(7777),
+				Path: "/local/Status",
+			}
+		}
+	}
+
+	if pmmSpec.ReadinessProbes != nil {
+		container.ReadinessProbe = pmmSpec.ReadinessProbes
+		if reflect.DeepEqual(container.ReadinessProbe.ProbeHandler, corev1.ProbeHandler{}) {
+			container.ReadinessProbe.HTTPGet = &corev1.HTTPGetAction{
+				Port: intstr.FromInt32(7777),
+				Path: "/local/Status",
+			}
+		}
+	}
+
+	return container
```

Comment:
I think we should add some new assertions to the existing unit test that we have for pmm.

Reply:
since we already have test for PMM container, i think it makes sense to check these in there